### PR TITLE
[docs]: update roformer.md model card

### PR DESCRIPTION
**Pull request by KsuParkhamchuk**
# What does this PR do?
Updates the RoFormer card according to the description in #https://github.com/huggingface/transformers/issues/36979 to standardize all model cards' look.
The pr includes:
- Model description
- Examples for Pipeline, AutoModel, and command line usage

Note: Quantization is not applied because of the model's small size
Note: AttentionMaskVisualizer is not applicable either, as RoFormer is currently not supported

- [x] This PR improves the docs 

## Who can review?

@stevhliu Please have a look and let me know if any further work is needed. thank you

### CODE REVIEW

File: docs/source/en/model_doc/roformer.md [hunk 1]
```diff
@@ -14,172 +14,174 @@ rendered properly in your Markdown viewer.
...
 ">
+    </div>
 </div>
 
-## Overview
+# RoFormer
+
+[RoFormer](https://huggingface.co/papers/2104.09864) is an enhansed transformer model with rotary position embeddings applied to the self-attention mechanism (particularly to Query and Key matrices). The purpose of the ReFormer is to represent the benefits of rotary position encoding (RoPE). Benefits include natural incorporation of the relative position and decaying inter-token dependencies (due to being based on complex number ideas), flexibility in sequence length, faster convergence during training, as well as an ability to be integrated into a linear self-attention mechanism.
```
Comment (stevhliu) [inline docs/source/en/model_doc/roformer.md:28]:
```suggestion
[RoFormer](https://huggingface.co/papers/2104.09864) introduces Rotary Position Embedding (RoPE) to encode token positions by rotating the inputs in 2D space. This allows a model to track absolute positions and model relative relationships. RoPE can scale to longer sequences, account for the natural decay of token dependencies, and works with the more efficient linear self-attention.
```

File: docs/source/en/model_doc/roformer.md [hunk 2]
```diff
@@ -14,172 +14,174 @@ rendered properly in your Markdown viewer.
...
+
+The RoFormer was originally evaluated on various NLP tasks including both English and Chineese datasets. Some of them mentioned in the paper include machine translation, performance during pre-training, fine-tuning on GLUE tasks and dealing with long texts showed RoFormer outperforming the chosen baseline models like BERTDelvin, model Vaswani et al.[2017], WoBERT Su and others.
+
+You can find all the original [RoFormer](link) checkpoints under the [RoFormer](link) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/roformer.md:32]:
```suggestion
You can find all the RoFormer checkpoints on the [Hub](https://huggingface.co/models?search=roformer).
```

File: docs/source/en/model_doc/roformer.md [hunk 3]
```diff
@@ -14,172 +14,174 @@ rendered properly in your Markdown viewer.
...
+
+> [!TIP]
+> Click on the [RoFormer] models in the right sidebar for more examples of how to apply [RoFormer] to different NLP tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/roformer.md:35]:
```suggestion
> Click on the RoFormer models in the right sidebar for more examples of how to apply RoFormer to different language tasks.
```

File: docs/source/en/model_doc/roformer.md [hunk 4]
```diff
@@ -14,172 +14,174 @@ rendered properly in your Markdown viewer.
...
+```py
+echo -e "水在零度时会[MASK]" | transformers run --task fill-mask --model junnyu/roformer_chinese_base --device 0
+```
+
+</hfoption>
+</hfoptions>
+
+## Notes
+
+- The current implementation of the RoFormer is an encoder-only model. The original code can be found [here](https://github.com/ZhuiyiTechnology/roformer)
```
Comment (stevhliu) [inline docs/source/en/model_doc/roformer.md:86]:
```suggestion
- The current RoFormer implementation is an encoder-only model. The original code can be found in the [ZhuiyiTechnology/roformer](https://github.com/ZhuiyiTechnology/roformer) repository.
```

File: docs/source/en/model_doc/roformer.md [hunk 5]
```diff
@@ -14,172 +14,174 @@ rendered properly in your Markdown viewer.
...
 [[autodoc]] RoFormerConfig
 
 ## RoFormerTokenizer
 
-[[autodoc]] RoFormerTokenizer
-    - build_inputs_with_special_tokens
-    - get_special_tokens_mask
-    - create_token_type_ids_from_sequences
-    - save_vocabulary
+[[autodoc]] RoFormerTokenizer - build_inputs_with_special_tokens - get_special_tokens_mask - create_token_type_ids_from_sequences - save_vocabulary
```
Comment (stevhliu) [inline docs/source/en/model_doc/roformer.md:94]: You can undo all the changes to the `[[autodoc]]` below.

File: docs/source/en/model_doc/roformer.md [hunk 6]
```diff
@@ -14,172 +14,174 @@ rendered properly in your Markdown viewer.
...
-transformer-based language models and propose a novel implementation named Rotary Position Embedding(RoPE). The
-proposed RoPE encodes absolute positional information with rotation matrix and naturally incorporates explicit relative
-position dependency in self-attention formulation. Notably, RoPE comes with valuable properties such as flexibility of
-being expand to any sequence lengths, decaying inter-token dependency with increasing relative distances, and
-capability of equipping the linear self-attention with relative position encoding. As a result, the enhanced
-transformer with rotary position embedding, or RoFormer, achieves superior performance in tasks with long texts. We
-release the theoretical analysis along with some preliminary experiment results on Chinese data. The undergoing
-experiment for English benchmark will soon be updated.*
+```py
+import torch
```
Comment (stevhliu) [inline docs/source/en/model_doc/roformer.md:43]:
```suggestion
# uncomment to install rjieba which is needed for the tokenizer
# !pip install rjieba
import torch
```

File: docs/source/en/model_doc/roformer.md [hunk 7]
```diff
@@ -14,172 +14,174 @@ rendered properly in your Markdown viewer.
...
 
-## Usage tips
-RoFormer is a BERT-like autoencoding model with rotary position embeddings. Rotary position embeddings have shown 
-improved performance on classification tasks with long texts.
+</hfoption>
+<hfoption id="AutoModel">
 
-## Resources
+```py
+import torch
```
Comment (stevhliu) [inline docs/source/en/model_doc/roformer.md:60]:
```suggestion
# uncomment to install rjieba which is needed for the tokenizer
# !pip install rjieba
import torch
```

**Review by stevhliu — COMMENTED** [review]
Nice work, thanks for your contribution!

**Comment by KsuParkhamchuk** [comment]
Thank you for the review @stevhliu ! The comments were addressed

**Review by stevhliu — APPROVED** [review]
Thanks, just a few more minor changes!

File: docs/source/en/model_doc/roformer.md [hunk 8]
```diff
@@ -72,9 +104,6 @@ improved performance on classification tasks with long texts.
 [[autodoc]] RoFormerTokenizerFast
     - build_inputs_with_special_tokens
 
-<frameworkcontent>
-<pt>
-
```
[the remark this answers is not in the document]
Comment (KsuParkhamchuk) [reply docs/source/en/model_doc/roformer.md]: moved them back
